app.ts

```diff
@@ -0,0 +1,323 @@
+			func(...args);
+		}, delay);
+	}
+}
+
+// Display records on the page based on the specified range.
+async function displayData(fromRecord: number, recordsDisplayed: number) {
+	$("#loader").show();
+	$("#tableWrapper").hide();
+	let isSearchMode = state.searchedIndex !== null;
+	const searchIndexOnPage = isSearchMode ? state.searchedIndex! - fromRecord : 0;
+	fromRecord = Math.max(fromRecord, 1);
+	const recordCount = await totalRecords();
+	const maximumRecords = Math.min(recordCount - fromRecord, recordsDisplayed);
+	const toRecord = fromRecord + maximumRecords - 1;
+	const data = await fetchRecords(fromRecord, toRecord);
+	let tableData = "";
+	if (isSearchMode) {
+		if (currentPage * state.recordsPerPage > recordCount) {
+			recordsDisplayed = recordCount - fromRecord;
+		}
+		state.currentValueOfFirstRecord = parseInt(data[0][0]);
+		state.currentFirstRecordIndex = fromRecord;
+		if (recordsDisplayed >= 16) {
+			currentPage = totalPages;
+			state.currentFirstRecordIndex = (totalPages - 1) * state.recordsPerPage;
+		}
+	} else {
+		fromRecord = Math.max(
+			fromRecord + searchIndexOnPage - (recordsDisplayed - 1), 0);
+	}
+	for (const record of data) {
+		tableData += "<tr>";
+		for (const value of record) {
+			tableData += `<td>${value}</td>`;
+		}
+		tableData += "</tr>";
+	}
+	$("#tableBody").html(tableData);
+	$("#loader").hide();
+	$("#tableWrapper").show();
+}
+
+// Search for records by a given value and display them on the page.
+async function searchMethod(searchValue: any) {
+	searchValue = Math.min(searchValue, (await totalRecords()) - 1);
+	let targetPage = Math.ceil((searchValue + 1) / state.recordsPerPage);
+	targetPage = Math.min(targetPage, totalPages);
+	const lastRecordIndex = (await totalRecords()) - 1;
+	const searchIndex = Math.min(searchValue, lastRecordIndex);
+	const searchPage = Math.ceil((searchIndex + 1) / state.recordsPerPage);
+	const fromRecord = Math.max(searchIndex - (state.recordsPerPage - 1), 0);
+	const toRecord = Math.min(fromRecord + state.recordsPerPage - 1, lastRecordIndex);
+	return fetchRecords(fromRecord, toRecord)
+		.then((records) => {
+			let foundIndex = -1;
+			let recordIndex = 0;
+			for (const record of records) {
+				const idValue = parseInt(record[0]);
+				if (idValue === searchValue) {
+					foundIndex = fromRecord + recordIndex;
+					break;
+				}
+				recordIndex++;
+			}
+			if (foundIndex !== -1) {
+				state.searchedIndex = foundIndex;
+				currentPage = searchPage;
+				state.currentFirstRecordIndex = (searchPage - 1) * state.recordsPerPage;
+				return displayData(state.currentFirstRecordIndex, state.recordsPerPage);
+			} else {
+				state.searchedIndex = null;
+			}
+			return Promise.resolve();
+		})
+		.then(() => {
+			$("#loader").hide();
+			$("#tableWrapper").show();
+			$("#prevPageButton").show();
+			if (searchPage === totalPages) {
+				$("#nextPageButton").hide();
+			} else {
+				$("#nextPageButton").show();
+			}
+		}).catch((error) => {
```

Comment:
```suggestion
		})
		.catch((error) => {
```
Put catches on the next line like the suggestion. The same like you did with the thens